File: src/js/agent.ts
/**
 * Agent framework
 */

export interface AgentOptions {
  name: string;
  model?: string;
  systemPrompt?: string;
  tools?: string[];
}

export class Agent {
  name: string;
  model: string;
  systemPrompt?: string;
  tools: string[];

  constructor(options: AgentOptions) {
    this.name = options.name;
    this.model = options.model || 'gpt-4';
    this.systemPrompt = options.systemPrompt;
    this.tools = options.tools || [];
  }

  async run(task: string): Promise<string> {
    // Implementation would execute the agent
    return `Task '${task}' completed by ${this.name}`;
  }

  addTool(tool: string): void {
    this.tools.push(tool);
  }

  toJSON(): any {
    return {
      name: this.name,
      model: this.model,
      systemPrompt: this.systemPrompt,
      tools: this.tools
    };
  }
}

File: src/rs/src/agent.rs
//! Agent framework

use anyhow::Result;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Agent {
    pub name: String,
    pub model: String,
    pub system_prompt: Option<String>,
    pub tools: Vec<String>,
}

impl Agent {
    pub fn new(name: String) -> Self {
        Self {
            name,
            model: "gpt-4".to_string(),
            system_prompt: None,
            tools: Vec::new(),
        }
    }

    pub async fn run(&self, task: &str) -> Result<String> {
        Ok(format!("Task '{}' completed by {}", task, self.name))
    }

    pub fn add_tool(&mut self, tool: String) {
        self.tools.push(tool);
    }
}